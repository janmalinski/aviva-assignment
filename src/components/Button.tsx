import { StyleSheet, Text, TouchableOpacity } from 'react-native'
import React from 'react'

interface IButtonProps {
    title:  string;
    onPress: () => void;
    testID?: string;
};

const Button = ({title, onPress, testID}: IButtonProps) => {
  return (
    <TouchableOpacity testID={testID} style={styles.button} onPress={onPress}>
      <Text>{title}</Text>
    </TouchableOpacity>
  )
}

export default Button;

const styles = StyleSheet.create({
    button: {
        borderWidth: 1,
        borderRadius: 4,
        paddingHorizontal: 15,
        alignItems: 'center',
        justifyContent: 'center',
        minHeight: 56,
    }
})