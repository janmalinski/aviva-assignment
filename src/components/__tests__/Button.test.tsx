import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import Button from '../Button';

const title = 'Back';
const onPress = jest.fn();


describe('Rendering Button', () => {
    const ButtonEl = (
        <Button testID='button' title={title} onPress={onPress}/>
    );

    it('Should match a snapshot', () => {
        const elJson = render(ButtonEl).toJSON();
        expect(elJson).toMatchSnapshot(); 
    });
});

describe('Behaviour Button', () => {
   
    it('Should press  button', () => {
        const { getByTestId } = render(<Button testID='button' title={title} onPress={onPress}/>);
        const button = getByTestId('button');
        expect(button).toBeTruthy();
        fireEvent.press(button);
    });
});
