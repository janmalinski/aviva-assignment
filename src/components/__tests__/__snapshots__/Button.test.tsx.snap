// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Button Should match a snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 4,
      "borderWidth": 1,
      "justifyContent": "center",
      "minHeight": 56,
      "opacity": 1,
      "paddingHorizontal": 15,
    }
  }
  testID="button"
>
  <Text>
    Back
  </Text>
</View>
`;
